docker-java-core/src/main/java/com/github/dockerjava/core/DefaultDockerClientConfig.java

```diff
@@ -443,14 +439,33 @@ public DefaultDockerClientConfig build() {
                 sslConfig = customSslConfig;
             }
 
+            final DockerConfigFile dockerConfigFile = readDockerConfig();
+
             URI dockerHostUri = dockerHost != null
                 ? dockerHost
-                : URI.create(SystemUtils.IS_OS_WINDOWS ? WINDOWS_DEFAULT_DOCKER_HOST : DEFAULT_DOCKER_HOST);
+                : dockerHostFromContextOrDefault(dockerConfigFile);
 
-            return new DefaultDockerClientConfig(dockerHostUri, dockerConfig, apiVersion, registryUrl, registryUsername,
+            return new DefaultDockerClientConfig(dockerHostUri, dockerConfigFile, dockerConfig, apiVersion, registryUrl, registryUsername,
                     registryPassword, registryEmail, sslConfig);
         }
 
+        private DockerConfigFile readDockerConfig() {
+            try {
+                return DockerConfigFile.loadConfig(DockerClientConfig.getDefaultObjectMapper(), dockerConfig);
+            } catch (IOException e) {
+                throw new DockerClientException("Failed to parse docker configuration file", e);
+            }
+        }
+
+        private URI dockerHostFromContextOrDefault(DockerConfigFile dockerConfigFile) {
+            final String currentContext = dockerConfigFile.getCurrentContext();
```

Comment:
current context can be obtained via `DOCKER_CONTEXT` env var or reading `currentContext` from `~/.docker/config.json`.

Reply:
I have now added support for `DOCKER_CONTEXT` env var.